src/renderer/components/cluster-manager/cluster-view.route.ts

```diff
@@ -16,12 +15,12 @@ export const clusterViewRoute: RouteProps = {
 
 export const clusterViewURL = buildURL<IClusterViewRouteParams>(clusterViewRoute.path)
 
-export function getMatchedClusterId(): string {
+export function getMatchedClusterId(extraRoutes: RouteProps[] = []): string {
```

Comment:
Let's not overcomplicate this function. It serves only single purpose - get current cluster-id from cluster-view.

Anyway `clusterSettingsRoute.path` can be removed safely and `cluster-id` could be obtained from `this.props.match.params` in `ClusterSettings` since it's a `Route` component.

In general if this would be not enough I would recommend to change logic in `windowManager.reload`
"If a cluster has frame-id then always reload that frame" looks wrong IMHO..